packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/from/V1_MappingTransformer.ts

```diff
@@ -463,7 +467,11 @@ const transformRelationalPropertyMapping = (
           [SOURCE_INFORMATION_KEY],
         ),
   ) as V1_RawRelationalOperationElement;
-  propertyMapping.source = undefined; // @MARKER: GRAMMAR ROUNDTRIP --- omit this information during protocol transformation as it can be interpreted while building the graph
+  // Note: renderSourceId is required to have the source information persisted in case of relationalPropertyMappings
+  // source key needs to be interpeted here for consistency between engine protcol and studio protcol
+  propertyMapping.source = renderSourceId
+    ? transformPropertyMappingSource(element.sourceSetImplementation)
+    : undefined; // @MARKER: GRAMMAR ROUNDTRIP --- need this filed to render both source and target field in roundetrip Grammar
```

Comment:
Let's keep the comment on the marker the same as everywhere else and move this specific comment to above/merge it with the `NOTE` section